### DISCUSSION TITLE
range selection doesn't work with only 'to' date

### DISCUSSION THREAD

**Discussion by Sam-Apostel** [gpbl/react-day-picker#1875] [Support]
When using `mode="range"` you can use `selected={{ from: new Date(), to: undefined }}` and it renders correctly. But when I use `selected={{ from: undefined, to: new Date() }}` nothing seems to be selected and the next time you click a date the `onSelect` callback doesn't receive the `to`-date.

Is this a bug or intentional?
Is there a way to toggle this behaviour? What would this look like if I were to implement this?

**Comment by Sam-Apostel** [comment]
PR adding this behaviour: https://github.com/gpbl/react-day-picker/pull/1876

**Comment by dimitur2204** [comment]
@Sam-Apostel I believe this is a desired behaviour! Think about it - it makes sense to have a selected **from** date where the picker is clicked once with an option to pick an **end**, but not the other way around. 
In that case you have
```
{
    start: Date
    end: undefined
}
```
Try it here https://react-day-picker.js.org/basics/selecting-days#selecting-a-range-of-days.
